src/main/java/com/technicjelle/Cheese.java

```diff
@@ -173,20 +174,24 @@ public static Cheese createFromCellsUnsafe(Vector2d cellSize, Vector2i... cells)
 	 * @see #createMultiCheeseFromChunks(Vector2i...)
 	 */
 	public static boolean checkConnected(Vector2i... cells) {
-		Set<Vector2i> cellsToCheck = new HashSet<>(List.of(cells));
+		if (cells.length == 0) return false;
+		if (cells.length == 1) return true;
+
+		Set<Vector2i> cellsToCheck = Set.of(cells);
 		Set<Vector2i> visited = new HashSet<>();
-		Queue<Vector2i> toVisit = new LinkedList<>();
-		toVisit.add(cellsToCheck.iterator().next()); // start with the first cell
+		Stack<Vector2i> toVisit = new Stack<>();
+		toVisit.push(cells[0]); // start with the first cell
+		visited.add(cells[0]);
 
 		while (!toVisit.isEmpty()) {
-			Vector2i current = toVisit.poll();
-			visited.add(current);
+			Vector2i current = toVisit.pop();
 
 			// add all neighbours that are in the cellsToCheck set and not visited yet
 			for (Direction direction : Direction.values()) {
```

Comment:
If this one is eight-axis, might be worth checking out the performance impact of 4-axis expansion

Reply:
This is only four-axis.